pulsar/producer_partition.go

```diff
@@ -494,14 +496,15 @@ func (p *partitionProducer) internalSend(request *sendRequest) {
 
 	// The block chan must be closed when returned with exception
 	defer request.stopBlock()
-	if !p.canAddToQueue(request) {
+	if !p.canAddToQueue(request, uncompressedPayloadSize) {
 		return
 	}
 
 	if p.options.DisableMultiSchema {
 		if msg.Schema != nil && p.options.Schema != nil &&
 			msg.Schema.GetSchemaInfo().hash() != p.options.Schema.GetSchemaInfo().hash() {
 			p.publishSemaphore.Release()
+			p.client.memLimit.ReleaseMemory(uncompressedPayloadSize)
 			request.callback(nil, request.msg, fmt.Errorf("msg schema can not match with producer schema"))
 			p.log.WithError(err).Errorf("The producer %s of the topic %s is disabled the `MultiSchema`", p.producerName, p.topic)
```

Comment:
How about we combine these lines into one function like `completeCallbackAndReleaseSemaphore` in java? This will help subsequent developers not miss releasing any resources when adding new logic.